src/eva/data/mon_data_space.py

```diff
@@ -200,87 +196,279 @@ def subset_coordinate(self, ds, coordinate, requested_subset):
+
+                if line.find('vars') != -1:
                     strs = line.split()
-                    chans.append(strs[4])
+                    for st in strs:
+                        if st.isdigit():
+                            nvars = int(st)
 
                 if line.find('title') != -1:
+                    if line.find('conventional') == -1 and line.find('gsistat') == -1:
+                        strs = line.split()
+                        attribs['sensor'] = strs[1]
+                        attribs['sat'] = strs[2]
+
+                # Note we need to extract the actual channel numbers.  We have the
+                # number of channels via the xdef line, but they are not necessarily
+                # ordered consecutively.
+                if line.find('channel=') != -1:
                     strs = line.split()
-                    sensor = strs[1]
-                    satellite = strs[2]
+                    if strs[4].isdigit():
+                        channo.append(int(strs[4]))
 
-            channo = np.array(chans, dtype=int)
-        return channo, nchans, nregion, satellite, sensor
+            # The list of variables is at the end of the file between the lines
+            # "vars" and "end vars".
+            start = len(lines) - (nvars + 1)
+            for x in range(start, start + nvars):
+                strs = lines[x].split()
+                vars.append(strs[-1])
 
-    # ----------------------------------------------------------------------------------------------
+        return coords, dims, attribs, nvars, vars, channo
 
-    def read_radmon_ieee(self, file_name, nchans, nregion, file_path=None):
+    # ----------------------------------------------------------------------------------------------
 
-        if file_path:
-            filename = os.path.join(file_path, file_name)
-        else:
-            filename = file_name
-        if not os.path.isfile(filename):
-            count = None
-            penalty = None
-            omgnbc_sum = None
-            omgnbc_sum = None
-            total_sum = None
-            omgbc_sum = None
-            omgnbc_sum2 = None
-            omgbc_sum2 = None
-            total_sum2 = None
-            cycle_tm = None
-            return count, penalty, omgnbc_sum, omgnbc_sum, total_sum, \
-                omgbc_sum, omgnbc_sum2, omgbc_sum2, cycle_tm
+    def read_ieee(self, file_name, coords, dims, ndims_used, nvars, vars, file_path=None):
 
         # find cycle time in filename and create cycle_tm as datetime object
         cycle_tm = None
-        cycstrs = filename.split('.')
+        cycstrs = file_name.split('.')
 
         for cycstr in cycstrs:
             if cycstr.isnumeric():
                 cycle_tm = datetime(int(cycstr[0:4]), int(cycstr[4:6]),
                                     int(cycstr[6:8]), int(cycstr[8:]))
 
+        filename = os.path.join(file_path, file_name) if file_path else file_name
+
+        if not os.path.isfile(filename):
```

Comment:
What about printing a warning (or error) about missing filename using the logger?

Reply:
I'll add a warning.  It's not uncommon for there to be missing files over a given time span.  Whatever scripting solution I/we use to create the yaml files should step over the gaps and give the yaml a clean list, but the bottom line is one missing file isn't a fatal condition.

Reply:
Ok that's fine but yes something like `logging.info(f"File {filename} is missing")` should be sufficient